src/patternfly/components/Masthead/masthead.scss

```diff
@@ -127,19 +157,91 @@ $pf-c-masthead--inset-map: build-spacer-map("none", "sm", "md", "lg", "xl", "2xl
     --pf-c-masthead--BackgroundColor: var(--pf-c-masthead--m-light-200--BackgroundColor);
     --pf-c-masthead__main--BorderBottomColor: var(--pf-c-masthead--m-light-200__main--BorderBottomColor);
   }
+
+  .pf-c-toolbar {
+    --pf-c-toolbar--PaddingTop: var(--pf-c-masthead--c-toolbar--PaddingTop);
+    --pf-c-toolbar--PaddingBottom: var(--pf-c-masthead--c-toolbar--PaddingBottom);
+
+    width: 100%;
+  }
+
+  .pf-c-toolbar__content-section {
+    flex-wrap: nowrap;
+    min-width: 0;
+  }
+
+  .pf-c-toolbar__group,
+  .pf-c-toolbar__item {
+    min-width: 0;
+  }
+
+  // Context selectors and dropdowns
+  .pf-c-context-selector,
+  .pf-c-dropdown {
+    display: flex;
+    align-self: stretch;
+    height: 100%;
+  }
+
+  .pf-c-dropdown__toggle,
+  .pf-c-context-selector__toggle {
+    height: 100%;
+  }
+
+  .pf-c-dropdown__toggle.pf-m-plain {
+    align-self: center;
+    height: auto;
+  }
+
+  .pf-c-dropdown__toggle {
+    --pf-c-dropdown__toggle--before--BorderTopColor: transparent;
+    --pf-c-dropdown__toggle--before--BorderRightColor: transparent;
+    --pf-c-dropdown__toggle--before--BorderLeftColor: transparent;
+    --pf-c-dropdown--m-expanded__toggle--before--BorderBottomColor: var(--pf-c-masthead--c-dropdown__toggle--expanded--before--BorderBottomColor);
+    --pf-c-dropdown--m-expanded__toggle--before--BorderBottomWidth: var(--pf-c-masthead--c-dropdown__toggle--expanded--before--BorderBottomWidth);
+  }
+
+  .pf-c-dropdown__toggle::before {
+    border-width: 0;
+    border-bottom-width: var(--pf-c-masthead--c-context-selector__toggle--expanded--before--BorderBottomWidth);
+  }
+
+  .pf-c-context-selector {
+    --pf-c-context-selector--Width: auto;
+  }
+
+  .pf-c-context-selector__toggle {
+    --pf-c-context-selector__toggle--BorderTopColor: transparent;
+    --pf-c-context-selector__toggle--BorderRightColor: transparent;
+    --pf-c-context-selector__toggle--BorderLeftColor: transparent;
+    --pf-c-context-selector__toggle--expanded--BorderBottomColor: var(--pf-c-masthead--c-context-selector__toggle--expanded--before--BorderBottomColor);
+  }
+
+  .pf-c-context-selector__toggle::before {
+    border-width: 0;
+    border-bottom-width: var(--pf-c-masthead--c-context-selector__toggle--expanded--before--BorderBottomWidth);
+  }
+
```

Comment:
With this approach, we'll need to maintain a list of components/elements that need to be modified here. There are a lot of specific styles for those components being restyled in the masthead. I'm wondering if it makes more sense to maintain a variation of those components instead?

Also this will apply to all dropdowns/context selectors in the masthead - maybe that's good? Or maybe we should target these a little more specifically? I'd be interested to hear your thoughts on that.